dateutil/test/test_rrule.py

```diff
@@ -431,6 +431,28 @@ def testMonthlySkipForward(self):
                           datetime(2015, 3, 1, 0, 0),
                           datetime(2015, 3, 31, 0, 0)])
 
+    def testMonthlyByMonthSkipBackward(self):
```

Comment:
Does the new stuff cover the `bymonthday=30` case as well? Can you add tests for that?

Reply:
Not yet, in fact this was a fix for BYMONTH. But I've fixed it for BYMONTHDAY now.

Now I don't know how to deal with the negatives cases (do we also interpret FORWARD and BACKWARD in negative sense or not).